lib/opensearch_config.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Configuration module for OpenSearch feature flag
+# Caches the OPENSEARCH_ENABLED environment variable to avoid repeated lookups
+module OpenSearchConfig
+  def self.enabled?
+    @enabled ||= ENV['OPENSEARCH_ENABLED'] == 'true'
+  end
+
+  # Reset the cached value (useful for testing)
+  def self.reset!
+    @enabled = nil
+  end
+
+  # Returns the OpenSearch analytics client or raises an error if not initialized
+  def self.analytics_client
+    unless defined?(OPENSEARCH_ANALYTICS_CLIENT)
```

Comment:
Both analytics_client and search_client raise errors claiming "OPENSEARCH_ENABLED is true" without actually checking if OpenSearch is enabled. This creates misleading error messages when the constants are undefined but the feature is disabled

Maybe we should do something like this:
```suggestion
  def self.analytics_client
    return nil unless enabled?
    unless defined?(OPENSEARCH_ANALYTICS_CLIENT)
```